src/snoowrap.js

```diff
@@ -639,6 +639,14 @@ const snoowrap = class snoowrap {
     return this._get({uri: 'api/saved_categories'}).get('categories');
   }
 
+  /**
+   * @param {string} query a query to search
+   * @returns {Promise} A promis with array os subreddits starting with that query
+   */
+  searchSubreddit (query) {
+    return this._post({uri:'api/search_subreddits', form: {query}})
```

Comment:
The endpoint supports a few fields which could be added as parameters:
  * `exact`, `include_over_18`, `include_unadvertisable`, `query`

`query` also has a max length of 50 which could be checked for.